apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -460,6 +472,14 @@ type Storage struct {
 	// Whether this disk represents rotational storage
 	Rotational bool `json:"rotational,omitempty"`
 
+	// Device type, one of: HDD, SSD, NVME.
+	// When "Rotational" is "true", field value is set to "HDD".
+	// When "Rotational" is "false" and "Name" starts with "/dev/nvme",
+	// field value is set to "NVME".
+	// When previous cases are not covered, field value is set to "SSD".
+	// +kubebuilder:validation:Enum=HDD;SSD;NVME
```

Comment:
Add UNKNOWN aka empty value?

Reply:
Hmm, missed that due to broken controller-tools, will fix, thanks

Reply:
fixed